pennylane/beta/plugins/default_tensor.py

```diff
@@ -18,288 +18,46 @@
-            ],
-            [
-                0,
-                0,
-                np.exp(-1j * (a - c) / 2) * np.sin(b / 2),
-                np.exp(1j * (a + c) / 2) * np.cos(b / 2),
-            ],
-        ]
-    )
-
-
-# ========================================================
-#  Arbitrary states and operators
-# ========================================================
-
-
-def unitary(*args):
-    r"""Input validation for an arbitary unitary operation.
-
-    Args:
-        args (array): square unitary matrix
-
-    Raises:
-        ValueError: if the matrix is not unitary or square
-
-    Returns:
-        array: square unitary matrix
-    """
-    U = np.asarray(args[0])
-
-    if U.shape[0] != U.shape[1]:
-        raise ValueError("Operator must be a square matrix.")
-
-    if not np.allclose(U @ U.conj().T, np.identity(U.shape[0])):
-        raise ValueError("Operator must be unitary.")
-
-    return U
-
-
-def hermitian(*args):
-    r"""Input validation for an arbitary Hermitian expectation.
-
-    Args:
-        args (array): square hermitian matrix
-
-    Raises:
-        ValueError: if the matrix is not Hermitian or square
-
-    Returns:
-        array: square hermitian matrix
-    """
-    A = np.asarray(args[0])
-    if A.shape[0] != A.shape[1]:
-        raise ValueError("Expectation must be a square matrix.")
-
-    if not np.allclose(A, A.conj().T):
-        raise ValueError("Expectation must be Hermitian.")
-
-    return A
-
-
-def identity(*_):
-    """Identity matrix observable.
-
-    Returns:
-        array: 2x2 identity matrix
-    """
-    return np.identity(2)
-
-
-# ========================================================
-#  device
-# ========================================================
+contract_fns = {
+    "greedy": tn.contractors.greedy,
+    "branch": tn.contractors.branch,
+    "optimal": tn.contractors.optimal,
+    "auto": tn.contractors.auto,
+}
 
 
 class DefaultTensor(Device):
     """Experimental Tensor Network simulator device for PennyLane.
 
     Args:
-        wires (int): the number of modes to initialize the device in
+        wires (int): number of subsystems in the quantum state represented by the device
+        shots (int): Number of circuit evaluations/random samples to return when sampling from the device.
+            Defaults to 1000 if not specified.
+        representation (str): Underlying representation used for the tensor network simulation.
+            Valid options are "exact" (no approximations made) or "mps" (simulated quantum
+            state is approximated as a Matrix Product State).
+        contraction_method (str): Method used to perform tensor network contractions. Only applicable
+            for the "exact" representation. Valid options are "auto", "greedy", "branch", or "optimal".
+            See documentation of the `TensorNetwork library <https://tensornetwork.readthedocs.io/en/latest/>`_
+            for more information about contraction methods.
```

Comment:
I suggest adding an example here, as well as (optionally) more detailed examples under `.. UsageDetails::`. This way we can pass on this link when people have questions. See the [DefaultTensorTF](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.beta.plugins.DefaultTensorTF.html) docstring